File: 0001-two-sum/0001-two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int, int> mp ;
        int n = nums.size();
        vector<int> res(2);
        for (int i = 0; i < n; i++) {
            int find = target - nums[i];
           if(mp.find(find) != mp.end()){
               res[0] = mp[find];               
               res[1] = i;
           }
            else{
                mp[nums[i]] = i;
            }
        }
        return res;
    }
};

File: 0001-two-sum/0001-two-sum.py
# class Solution:
#     def twoSum(self, nums: List[int], target: int) -> List[int]:
#         n = len(nums)

#         for i in range(n):
#             for j in range(i+1, n):
#                 if nums[i] + nums[j] == target:
#                     return [i, j]
        
#         return []
        

class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        n = len(nums)
        res = []
        idx_map = {}
        
        for i in range(n):
            complement = target - nums[i]
            if complement in idx_map:
                res.extend([idx_map[complement], i])
                break
            else:
                idx_map[nums[i]] = i

        return res